packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -570,9 +552,7 @@ export function makeVatKeeper(
     if (snapStore) {
       snapStore.deleteVatSnapshots(vatID);
     }
-
-    const endPos = getRequired(`${vatID}.t.endPosition`);
-    kvStore.set(`${vatID}.t.startPosition`, endPos);
+    transcriptStore.rolloverSpan(vatID);
```

Comment:
This is the right thing to do, although we should eventually consider some cleanup and make the caller responsible for updating the transcript bounds. Or rename the function.

As @mhofman pointed out, this is used during vat upgrades, when we want to switch to a new incarnation, which means a new span boundary, and to stop using the previous snapshot. We call `deleteVatSnapshots` because previously that was the only way to stop using the latest snapshot. But now we've got this `isCurrent=0` thing, which lets us ignore without actually deleting.

Thinking about our new data retention behavior, I think once this PR lands, we'll:
* retain all transcript entries, until either someone runs an external `/usr/bin/sqlite3` command to delete them, or we add more code to prune old spans
  * even vat termination won't delete the transcript entries
* retain old snapshots, until:
  * the vat is terminated
  * the vat is upgraded

That's workable, but not entirely consistent. I think vat termination should delete everything: snapshots *and* transcripts. I think we'll need to make it easy for validators to turn on pruning of old snapshots.. pruning snapshots at upgrade points is helpful, but feels like an odd side-effect. Until we get the pruning config implemented, maybe we should mark the latest snapshot as inactive during vat upgrade, instead of deleting everything.

Reply:
Given that right now we only keep the latest snapshot of a vat, I believe the behavior is at least consistent within store types.

However I agree that in the future we should have a pruning policy that decides how many "spans" of a vat to keep, which would keep both the snapshot and transcript for that span consistently with each other. Most validators likely only care to have the latest span for restart, and there is no reason to keep historical transcript spans.